pkg/controller/jobframework/validation.go

```diff
@@ -71,3 +74,19 @@ func ValidateUpdateForParentWorkload(oldJob, newJob GenericJob) field.ErrorList
 	}
 	return allErrs
 }
+
+func ValidateUpdateForOriginalNodeSelectors(oldJob, newJob GenericJob) field.ErrorList {
```

Comment:
This is tested by the job and mpijob unit tests.

 Maybe it could make sense to have a GenericJob test double that could be used in unit tests for this package, however this should be done in a follow-up and should cover more that `ValidateUpdateForOriginalNodeSelectors`.